Firestore/core/src/firebase/firestore/util/to_string.h

```diff
@@ -0,0 +1,202 @@
+/*
+ * Copyright 2019 Google
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef FIRESTORE_CORE_SRC_FIREBASE_FIRESTORE_UTIL_TO_STRING_H_
+#define FIRESTORE_CORE_SRC_FIREBASE_FIRESTORE_UTIL_TO_STRING_H_
+
+#if !defined(__OBJC__)
+#error "This header only supports Objective-C++"
+#endif  // !defined(__OBJC__)
+
+#import <Foundation/Foundation.h>
+
+#include <algorithm>
+#include <numeric>
+#include <string>
+#include <type_traits>
+#include <utility>
+
+#include "Firestore/core/src/firebase/firestore/util/string_apple.h"
+#include "Firestore/core/src/firebase/firestore/util/string_format.h"
+#include "Firestore/core/src/firebase/firestore/util/type_traits.h"
+#include "absl/meta/type_traits.h"
+#include "absl/strings/str_join.h"
+
+namespace firebase {
+namespace firestore {
+namespace util {
+
+template <typename T>
+std::string ToString(const T& value);
+
+namespace impl {
+
+// Checks whether the given type `T` defines a member function `ToString`
+template <typename T, typename = absl::void_t<>>
+struct has_to_string : std::false_type {};
+template <typename T>
+struct has_to_string<T, absl::void_t<decltype(std::declval<T>().ToString())>>
+    : std::true_type {};
+
+// Fallback
+
+template <typename T>
+std::string ToStringDefault(const T& value) {
```

Comment:
You're mixing "ToString" as a prefix and as a suffix in these member names. Shouldn't these be `DefaultToString` and `CustomToString` to match the others below?

Reply:
Done.